lib/Driver/FineGrainedDependencyDriverGraph.cpp

```diff
@@ -108,6 +108,27 @@ ModuleDepGraph::Changes ModuleDepGraph::loadFromSourceFileDepGraph(
   return changes;
 }
 
+ModuleDepGraph::Changes ModuleDepGraph::loadFromSwiftModuleBuffer(
+    const Job *Cmd, llvm::MemoryBuffer &buffer, DiagnosticEngine &diags) {
+  FrontendStatsTracer tracer(
+      stats, "fine-grained-dependencies-loadFromSwiftModuleBuffer");
+  PrettyStackTraceStringAction stackTrace(
+      "loading fine-grained dependency graph from swiftmodule",
+      buffer.getBufferIdentifier());
+
+   Optional<SourceFileDepGraph> sourceFileDepGraph =
+      SourceFileDepGraph::loadFromSwiftModuleBuffer(buffer);
+  if (!sourceFileDepGraph)
+    return None;
+  registerJob(Cmd);
+  auto changes = integrate(*sourceFileDepGraph, buffer.getBufferIdentifier());
+  if (verifyFineGrainedDependencyGraphAfterEveryImport)
+    verify();
+  if (emitFineGrainedDependencyDotFileAfterEveryImport)
+    emitDotFileForJob(diags, Cmd);
+  return changes;
```

Comment:
Not sure how much, but I suspect that most of this can be factored with the load-from-swiftdeps case.